### PR TITLE
Add Azure CDN and Front Door front-end cache invalidation support

### PR DESCRIPTION
**Pull request by tm-kn**
This PR adds support for invalidating cache on two platforms:

* Azure CDN
* Azure Front Door

This is based on [wagtail-azure-cdn](https://github.com/torchbox/wagtail-azure-cdn) with multi site support taken out as it would only make this PR more complex.

I am not sure if this is a rich approach to have a massive `backends.py` file. Would it help to first create a PR that splits individual backends into separate files?

### CODE REVIEW

File: setup.py
Comment (kaedroho) [inline setup.py:52]: Could you version checks here?

File: wagtail/contrib/frontend_cache/backends.py
Comment (kaedroho) [inline wagtail/contrib/frontend_cache/backends.py:203]: We've tightened up the flake8 config recently so it won't allow three spaces here anymore, please could you rebase this on the latest master?

**Review by kaedroho — APPROVED** [review]
Nice work! Just a couple of minor things to do before merge.

> I am not sure if this is a rich approach to have a massive backends.py file. Would it help to first create a PR that splits individual backends into separate files?

Happy for it to be a big file for now, we can split it up after merge

File: setup.py
Comment (zerolab) [reply setup.py:52]: I would very much like for us to move for optional dependencies where possible.
i.e. if you need the Azure CDN backend you'd do `pip install wagtail[cdn-azure]` or similar?

This is probably a large discussion to have, so not a blocker on this per se

**Comment by kaedroho** [comment]
Rebased and made amendments

File: setup.py
Comment (kaedroho) [reply setup.py]: Yep, I agree. We probably want to do something similar for cloudflare, elasticsearch, boto3, etc. Shall we tackle all these in one go in a separate PR?

**Comment by lb-** [comment]
Flagging this for discussion at the next core team meeting.

**Comment by lb-** [comment]
Good to go, action for LB to rebase/fix conflicts and Merge in.

**Comment by tm-kn** [comment]
Just to let you know, I made changes to use the newer Azure SDK - torchbox/wagtail-azure-cdn#4

File: setup.py
Comment (lb-) [reply setup.py]: @kaedroho  / @zerolab  I have split this out to a new issue, please can either of you update with a bit more context https://github.com/wagtail/wagtail/issues/7753

Comment (lb-) [reply setup.py]: Note: the original issue raised has been resolved

**Comment by lb-** [comment]
@tm-kn thanks for this, it has been merged in.
As for https://github.com/torchbox/wagtail-azure-cdn/pull/4 -> I thought it was best to add additional versions support in a new PR, it would be awesome if you could raise a new one when you get the chance.

Thanks @kaedroho  for reviewing.

**Pull request merged**
